test/spawn-options.node.js

```diff
@@ -0,0 +1,360 @@
+/* eslint-env mocha */
+/* eslint max-nested-callbacks: ["error", 8] */
+'use strict'
+
+const series = require('async/series')
+const waterfall = require('async/waterfall')
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+const fs = require('fs')
+const isNode = require('detect-node')
+const hat = require('hat')
+const DaemonFactory = require('../src')
+const tempDir = require('../src/utils/tmp-dir')
+const JSIPFS = require('ipfs')
+
+const tests = [
+  { type: 'go' },
+  { type: 'js' },
+  { type: 'proc', exec: JSIPFS }
+]
+
+describe('Spawn options', () => {
+  tests.forEach((dfOpts) => describe(`${dfOpts.type}`, () => {
+    // TODO pick this from go-ipfs-dep, otherwise test will keep breaking
+    const VERSION_STRING = dfOpts.type === 'js'
+      ? `js-ipfs version: ${require('ipfs/package.json').version}`
+      : 'ipfs version 0.4.13'
+    let df
+
+    before(() => {
+      df = DaemonFactory.create(dfOpts)
+    })
+
+    // TODO: why does ipfsd-ctl polute the env variables?
```

Comment:
It used to set IPFS_PATH and other variables, its not the case anymore - we can remove that.